File: 백준/Silver/17087. 숨바꼭질 6/숨바꼭질 6.py
from sys import stdin

input = stdin.readline

n, s = map(int, input().split())  # n: 동생 수, s:수빈 위치
pos_l = list(map(lambda x: abs(s - x), map(int, input().split())))


def get_gcd(a, b):
    if a < b:
        a, b = b, a

    if b == 0:
        return a

    return get_gcd(b, a % b)


ans = pos_l[0]
for num in pos_l:
    ans = get_gcd(ans, num)

print(ans)

File: Java/백준/Silver/17087. 숨바꼭질 6/숨바꼭질 6.java
import java.util.*;
import java.io.*;

class Main {

    static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    static StringTokenizer st;

    static int N, S, GCD;
    static int[] arr;
    static int[] nums;

    public static void main(String[] args) throws IOException {

        st = new StringTokenizer(br.readLine());
        N = Integer.parseInt(st.nextToken());
        S = Integer.parseInt(st.nextToken());

        arr = new int[N];
        nums = new int[N];
        st = new StringTokenizer(br.readLine());
        for (int i = 0; i < N; i++) {
            arr[i] = Integer.parseInt(st.nextToken());
            nums[i] = Math.abs(arr[i] - S);
        }

        GCD = nums[0];
        for (int num : nums) {
            GCD = getGcd(GCD, num);
        }

        System.out.println(GCD);
    }

    private static int getGcd(int a, int b) {
        if (a < b) {
            int temp = a;
            a = b;
            b = temp;
        }
        while (b > 0) {
            int temp = a;
            a = b;
            b = temp % b;
        }

        return a;
    }
}